src/axom/quest/InOutOctree.hpp

```diff
@@ -60,3362 +65,1514 @@ namespace quest
+    /// Find a point from this segment within the expanded bounding box of the mesh
+    // We'll use the midpoint of the segment after clipping it against the bounding box
+    double pMin, pMax;
+    const bool intersects = primal::intersect(seg, expandedBB, pMin, pMax);
+    if(!intersects)
     {
-      TriVertIndices tvRel = m_octree.m_meshWrapper.triangleVertexIndices(tIdx);
-      for(int j = 0; j < tvRel.size(); ++j)
-      {
-        VertexIndex vIdx = tvRel[j];
-        BlockIndex vertBlock = m_octree.m_vertexToBlockMap[vIdx];
-        const InOutBlockData& leafData = m_octree[vertBlock];
-
-        // Check that this triangle is referenced here.
-        bool foundTriangle = false;
-        TriangleIndexSet leafTris = m_octree.leafTriangles(vertBlock, leafData);
-        for(int k = 0; !foundTriangle && k < leafTris.size(); ++k)
-        {
-          if(leafTris[k] == tIdx) foundTriangle = true;
-        }
-
-        SLIC_ASSERT_MSG(
-          foundTriangle,
-          fmt::format("Did not find triangle with index {} and vertices [{}] "
-                      "in block {} containing vertex {}"
-                      " \n\n *** "
-                      "\n Leaf data: {} "
-                      "\n\t Triangles in block? [{}] "
-                      "\n ***",
-                      tIdx,
-                      fmt::join(tvRel, ", "),
-                      vertBlock,
-                      vIdx,
-                      leafData,
-                      fmt::join(leafTris.begin(), leafTris.end(), ", ")));
-      }
+      continue;
     }
-  }
+    segmentPt = seg.at(0.5 * (pMin + pMax));
 
-  void checkBlockIndexingConsistency() const
-  {
-    // Check that internal blocks have no triangle / vertex
-    //       and leaf blocks satisfy the conditions above
-    SLIC_DEBUG(
-      "--Checking that internal blocks have no data, and that leaves satisfy "
-      "all PM conditions");
+    // Using a ray from query pt to point on this segment
+    // Find closest intersection to surface within cell inside this bounding box
+    CellIndex tIdx = MeshWrapper<DIM>::NO_CELL;
+    double minRayParam = std::numeric_limits<double>::infinity();
+    double minSegParam = std::numeric_limits<double>::infinity();
+    SpaceRay ray(queryPt, SpaceVector(queryPt, segmentPt));
 
-    const double bb_scale_factor = m_octree.m_boundingBoxScaleFactor;
+    QUEST_OCTREE_DEBUG_LOG_IF(
+      DEBUG_BLOCK_1 == leafBlk || DEBUG_BLOCK_2 == leafBlk,
+      fmt::format("Checking if pt {} is within block {} with data {}, "
+                  "ray is {}, segment point is {} on segment with index {}.",
+                  queryPt,
+                  leafBlk,
+                  leafData,
+                  ray,
+                  segmentPt,
+                  idx));
 
-    for(int lev = 0; lev < m_octree.m_levels.size(); ++lev)
+    double rayParam = 0;
+    double segParam = 0;
+    if(primal::intersect(ray, seg, rayParam, segParam))
```

Comment:
Similarly, this new formulation of `intersect(Ray, Segment)` returns the parametric coordinates of the intersection point. We need both of them to efficiently determine whether the query point is inside or outside